src/common/buffer.cc

```diff
@@ -2055,6 +2057,28 @@ sha1_digest_t buffer::list::sha1()
   return sha1_digest_t(fingerprint);
 }
 
+sha256_digest_t buffer::list::sha256()
```

Comment:
i'd suggest implement them as free functions. as they are not tightly coupled with `bufferlist`, instead they are basically the consumers of `list::buffers()` member function.

Reply:
All of the current hashing algorithms are bufferlist methods.  Making them freestanding would be fine, but we can do that as a separate change/cleanup?

Reply:
#28486